File: src/easy/387-first-unique-character-in-a-string/firstUniqueCharacterInAString.js
/**
 * @param {string} s
 * @return {number}
 */
var firstUniqChar = function(s) {
  const map = {};

  for (let i = 0; i < s.length; i++) {
    if (!map[s[i]]) {
      map[s[i]] = 1;
    } else {
      map[s[i]] = map[s[i]] + 1;
    }
  }

  for (let i = 0; i < s.length; i++) {
    if (map[s[i]] === 1) {
      return i;
    }
  }

  return -1;
};

console.log(firstUniqChar("leetcode"));        // output == 0
console.log(firstUniqChar("loveleetcode"));    // output == 2
console.log(firstUniqChar("aabb"));            // output == -1

File: src/easy/387-first-unique-character-in-a-string/firstUniqueCharacterInAString.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
      map = {}

      for x in s:
        if x in map:
          map[x] = map[x] + 1
        else:
          map[x] = 1

      for i in range(len(s)):
        if map[s[i]] == 1:
          return i

      return -1

testSolution = Solution()        

print(testSolution.firstUniqChar("leetcode"));        # output == 0
print(testSolution.firstUniqChar("loveleetcode"));    # output == 2
print(testSolution.firstUniqChar("aabb"));            # output == -1